File: frontend-main/src/models/rally/assignment.ts
export class Assignment {
  id: string;
  joiningCode: number;
  audience: Audience;

  constructor(id: string, joiningCode: number, audience: Audience) {
    this.id = id;
    this.joiningCode = joiningCode;
    this.audience = audience;
  }
}

export class Audience {
  id: string;
  title: string;
  participatingParties: ParticipatingParty[];

  constructor(
    id: string,
    title: string,
    participatingParties: ParticipatingParty[]
  ) {
    this.id = id;
    this.title = title;
    this.participatingParties = participatingParties;
  }
}

export class ParticipatingParty {
  id: string;
  title: string;

  constructor(id: string, title: string) {
    this.id = id;
    this.title = title;
  }
}

File: flutter-application-main/lib/models/rally/assignment.dart
import 'package:dart_mappable/dart_mappable.dart';

import 'audience.dart';

@MappableClass()
class Assignment {
  final String id;
  final int joiningCode;
  final Audience audience;

  Assignment({
    required this.id,
    required this.joiningCode,
    required this.audience,
  });
}
